plugins/cache/extract.go

```diff
@@ -80,6 +81,10 @@ func extractArchive(source io.Reader, target fileSystem) error {
 		} else if info.Mode().IsRegular() {
 			debug("extracting file: '%s'", header.Name)
 
+			if strings.HasPrefix(header.Name, "..") {
+				return fmt.Errorf("%s: illegal file", header.Name)
```

Comment:
`return runtime.NewMalformedPayloadError(` and this should be `if strings.hasPrefix(path.Clean(header.name)) {`
that way we'll also catch scenarios where we have `myfolder/../../../haha-evil.binary`